internal/pkg/manifest/transform.go

```diff
@@ -19,89 +42,281 @@ var basicKinds = []reflect.Kind{
 	reflect.Array, reflect.String, reflect.Slice,
 }
 
-type workloadTransformer struct{}
+type imageTransformer struct{}
 
-// Transformer returns custom merge logic for workload's fields.
-func (t workloadTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
-	if typ == reflect.TypeOf(Image{}) {
-		return transformImage()
+// Transformer provides custom logic to transform an Image.
+func (t imageTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
+	if typ != reflect.TypeOf(Image{}) {
+		return nil
 	}
 
-	if typ == reflect.TypeOf(EntryPointOverride{}) {
-		return transformStringSliceOrString(reflect.TypeOf(EntryPointOverride{}))
+	return func(dst, src reflect.Value) error {
+		// NOTE: Since `dst` comes from an exported field (all manifest fields are exported), `Interface{}` won't panic.
```

Comment:
This is more of a concern whether it's 
```go
type ImageOverride struct {
  EntryPoint EntryPointOverride
}
```
and not 
```go
type ImageOverride struct {
  entryPoint EntryPointOverride
}
```
`Interface()` panics if `dst` is obtained from an unexported field. This will never happen though, because `mergo` do not attempt to merge any unexported field.

Will remove the comment!